### PR TITLE
Turn off implicit casts and "avoid as" to prep for NNBD migration

### PR DESCRIPTION
**Pull request by dnfield**
This should result in no semantic/runtime differences,and is required for eventual null-safety migration.

### CODE REVIEW

**Comment by digiter** [comment]
Did you use some tool in generating these changes?

**Comment by dnfield** [comment]
Unfortunately it is not at all automated :)

**Comment by dnfield** [comment]
I changed analysis_options.yaml and then fixed all the call sites, sometimes using regexes to find and replace common things.

File: app_dart/lib/src/model/appengine/key_helper.dart [hunk 1]
```diff
@@ -157,9 +157,9 @@ class KeyHelper {
         }
         if (key.id != null) {
           if (key.id is String) {
-            element.name = key.id;
+            element.name = key.id as String;
           } else if (key.id is int) {
-            element.id = Int64(key.id);
+            element.id = Int64(key.id as int);
           }
         }
```
Comment (a14n) [inline app_dart/lib/src/model/appengine/key_helper.dart]: You can avoid the 2 casts by extracting a `id` local variable.

File: app_dart/lib/src/model/appengine/key_wrapper.dart [hunk 2]
```diff
@@ -43,9 +43,9 @@ class KeyWrapper {
       }
       if (slice.id != null) {
         if (slice.id is String) {
-          current.name = slice.id;
+          current.name = slice.id as String;
         } else if (slice.id is int) {
-          current.uid = Int64(slice.id);
+          current.uid = Int64(slice.id as int);
```
Comment (a14n) [inline app_dart/lib/src/model/appengine/key_wrapper.dart]: ditto

File: app_dart/lib/src/model/appengine/key_helper.dart (hunk 1 above)
Comment (dnfield) [reply app_dart/lib/src/model/appengine/key_helper.dart]: Done and done, thanks

Comment (a14n) [reply app_dart/lib/src/model/appengine/key_helper.dart]: Actually you can also remove the surrounding if-not-null

Comment (dnfield) [reply app_dart/lib/src/model/appengine/key_helper.dart]: That thought occurred to me and then for some reason I didn't do it.  Thanks!

**Pull request merged**